pinot-core/src/main/java/com/linkedin/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -90,82 +87,73 @@ public void init(Configuration configuration, DataManager dataManager, ServerMet
   }
 
   @Override
-  public DataTable processQuery(final ServerQueryRequest queryRequest, ExecutorService executorService) {
+  public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService) {
     TimerContext timerContext = queryRequest.getTimerContext();
     TimerContext.Timer schedulerWaitTimer = timerContext.getPhaseTimer(ServerQueryPhase.SCHEDULER_WAIT);
     if (schedulerWaitTimer != null) {
       schedulerWaitTimer.stopAndRecord();
     }
     TimerContext.Timer queryProcessingTimer = timerContext.startNewPhaseTimer(ServerQueryPhase.QUERY_PROCESSING);
 
-    DataTable dataTable;
-    List<SegmentDataManager> queryableSegmentDataManagerList = null;
     InstanceRequest instanceRequest = queryRequest.getInstanceRequest();
-    final long requestId = instanceRequest.getRequestId();
-
-    try {
+    long requestId = instanceRequest.getRequestId();
+    boolean enableTrace = instanceRequest.isEnableTrace();
+    if (enableTrace) {
       TraceContext.register(instanceRequest);
-      final BrokerRequest brokerRequest = instanceRequest.getQuery();
-      LOGGER.debug("Incoming query is : {}", brokerRequest);
+    }
+    BrokerRequest brokerRequest = instanceRequest.getQuery();
+    LOGGER.debug("Incoming request Id: {}, query: {}", requestId, brokerRequest);
+    String tableName = brokerRequest.getQuerySource().getTableName();
+    TableDataManager tableDataManager = _instanceDataManager.getTableDataManager(tableName);
+    Preconditions.checkState(tableDataManager != null, "Failed to find data manager for table: {}", tableName);
```

Comment:
All logic and pre-conditions should be inside the try/catch block, so that we return a correct dataTable to the broker and not an exception trace.

Reply:
We have another level of exception handling (unhandled exception) on the caller. On this level, this part should not throw exception